Client/Extensions/UIAlertControllerExtensions.swift

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+typealias UIAlertActionCallback = (UIAlertAction) -> Void
+
+// MARK: - Extension methods for building specific UIAlertController instances used across the app
+extension UIAlertController {
+
+    /**
+    Builds the Alert view that asks the user if they wish to opt into crash reporting.
+
+    - parameter sendReportCallback: Send report option handler
+    - parameter alwaysSendCallback: Always send option handler
+    - parameter dontSendCallback:   Dont send option handler
+    - parameter neverSendCallback:  Never send option handler
+
+    - returns: UIAlertController for opting into crash reporting after a crash occurred
+    */
+    class func crashOptInAlert(
+        sendReportCallback sendReportCallback: UIAlertActionCallback,
+        alwaysSendCallback: UIAlertActionCallback,
+        dontSendCallback: UIAlertActionCallback) -> UIAlertController {
+
+        let alert = UIAlertController(
+            title: NSLocalizedString("Oops! Firefox crashed", comment: "Title for prompt displayed to user after the app crashes"),
+            message: NSLocalizedString("Send a crash report so Mozilla can fix the problem?", comment: "Message displayed in the crash dialog above the buttons used to select when sending reports"),
+            preferredStyle: UIAlertControllerStyle.Alert
+        )
+
+        let sendReport = UIAlertAction(
+            title: NSLocalizedString("Send Report", comment: "Used as a button label for crash dialog prompt"),
```

Comment:
I think all of your added strings should have named tables. Ask @st3fan what the current style is for doing that.